src/main/java/com/playmonumenta/structures/StructuresAPI.java

```diff
@@ -697,24 +682,21 @@ private static void ensureTask(Plugin plugin) {
 			@Override
 			public void run() {
 				while (true) {
-					PendingTask task = PENDING_TASKS.pollFirst();
-
-					if (task != null) {
-						try {
-							Bukkit.getScheduler().runTask(plugin, task.mStartTask);
-							task.mSignal.get(45, TimeUnit.SECONDS);
-						} catch (Exception ex) {
-							plugin.getLogger().severe("Structure task took longer than 45s to complete! Continuing to the next task. Exception: " + ex.getMessage());
-						}
-					}
-
+					PendingTask task;
 					try {
-						Thread.sleep(50);
-					} catch (Exception ex) {
+						task = PENDING_TASKS.take();
+					} catch (InterruptedException ignored) {
 						plugin.getLogger().info("Structure loading task sleep was interrupted");
 						RUNNING_TASK = null;
 						break;
 					}
+
+					try {
+						Bukkit.getScheduler().runTask(plugin, task.mStartTask);
+						task.mSignal.get(45, TimeUnit.SECONDS);
+					} catch (Exception ex) {
+						plugin.getLogger().severe("Structure task took longer than 45s to complete! Continuing to the next task. Exception: " + ex.getMessage());
+					}
 				}
 			}
 		};
```

Comment:
I think this whole thing needs some restructuring, really weird setup right now. Need some timeout for the PENDING_TASKS.take() I think? 

Maybe we should just revert this entire block and move this change to another PR